### PR TITLE
UI fixes (#TheBraveLittleTailor)

### PR DESCRIPTION
**Pull request by OzoneH3**
Fixes: #21993 
Fixes: #21990 
Fixes: #21971 
Fixes: #21262 
Fixes: #20921 
Fixes: #20655 
Fixes: #20075 

New Character:
* Correctly color forbidden scenario traits.
* Fixed save template redraw and text query issue.
* Fix partly offscreen text in profession and scenario tab.
* Position profession info the same way as in scenario.

Craftin menu:
* Update requirements after filtering.

Layering menu:
* Prevent wrapping of 3 digit numbers.

Mod selection:
* Redraw mod text after removing a mod.

List items:
* Correct coloring when all items are on -priority.

Other:
* Actually draw the text from intro() on too small terminal windows.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
First of the "Fixes" leads to some PR instead of issue.

**Comment by OzoneH3** [comment]
> First of the "Fixes" leads to some PR instead of issue.

Fixed.

File: src/armor_layers.cpp [hunk 1]
```diff
@@ -393,8 +393,8 @@ void player::sort_armor()
                 if( rightListSize >= rightListOffset && pos <= cont_h - 2 ) {
                     trim_and_print( w_sort_right, pos, 2, right_w - 4, elem.damage,
                                     elem.name.c_str() );
-                    mvwprintz( w_sort_right, pos, right_w - 2, c_ltgray, "%d",
-                               elem.encumber );
+                    mvwprintz( w_sort_right, pos, right_w - 3, c_ltgray, "%*d",
+                               3, elem.encumber );
```
Comment (BevapDin) [inline src/armor_layers.cpp]: Out of curiosity: why the separate argument instead of putting the number into the format string as "%3d"?

Comment (OzoneH3) [reply src/armor_layers.cpp]: Ah didn't know it could be in there, took it from another place in the code that used *

**Pull request merged**